### ISSUE TITLE
[Discussion] Router & Babel upgrades to improve prerender capabilities

### ISSUE THREAD

**Issue by dac09** [redwoodjs/redwood#1670]
### Background
Currently, the router has been written for client side only, while Prerender Part 1 resolves this partially, there are some incremental changes that could make prerendering a lot more robust. 

The main issue is page loaders - with the combination of autoload routes babel plugin, and using "specs" to load pages, this makes it a significantly harder job running the router in Node.js

### Proposals

**1. Export LocationProvider from router**
Nice and easy, and don't think there's any reason not to do this.

This will allow us to control what path the route "sees" when window.location isn't availble

**2. Conditionally disable redwood-routes-autoloader plugin**
This plugin allows the sweet code-splitting-by-default capability on the clientside, but causes issues running in node

I can't seem to disable the autoloader plugin, in favour of a different one for prerender. How can we replace the plugin in redwood's babel preset?

**3. Change the router to render synchronously in prerender mode**
Dynamic imports are async, and must use `useEffect` or `componentDidMount` to resolve the page. Neither of these things run when rendering in Node, plus, we do not have access to modifying state when running componentWillMount (soon to be depracated anyway)

I can do this, but the code looks really horrible 

in packages/router/src/page-loader.js
```
.
.
  render() {
    if (global.__REDWOOD__PRERENDERING) {
      console.log('Page loader prerender block', this.props.spec)
      const PageFromLoader = this.props.spec.loader.default  // <--- ew

      return (
        <ParamsContext.Provider value={this.state.params}>
          <PageFromLoader />
        </ParamsContext.Provider>
      )
    }
```
Would love some suggestions here.

**Comment by peterp** [comment]
2. Conditionally disable redwood-routes-autoloader plugin

I left a comment in your repo. https://github.com/redwoodjs/redwood/pull/1641/files#r564785928

3. Change the router to render synchronously in prerender mode

What if we modified the way the babel plugin did the imports? We could write them to be none spec compliant, no async and then the router might be happy?

**Comment by dac09** [comment]
Closing this, changes implemented in #1641
